src/main/java/repository/CustomerRepository.java

```diff
@@ -0,0 +1,19 @@
+package repository;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import entity.Customer;
+
+@Repository
+public interface CustomerRepository extends JpaRepository<Customer, BigDecimal> {
+
+	@Cacheable("findByCustRep_AgeIsGreaterThan")
+	List<Customer> findByCustRep_AgeIsGreaterThan(BigDecimal age);
+
+	Customer findByCustNum(BigDecimal custNum);
```

Comment:
Repository include method findById. You dublicate it - this is bad

Reply:
Drop it from there - dublicate internal JpaRepository.findById. CustNum is Id field for customers